File: Pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //user input
    string text = get_string("Text: ");

    //get letters, words and sentances
    int textLen = strlen(text);
    int letters = 0;
    int words = 1;  //starts at one because there wont be a space at the end of the sentance
    int sentances = 0;
    
    //loop through the text to get data
    for (int i = 0; i < textLen; i++)
    {
        if (((int)text[i] >= 'a' && (int)text[i] <= 'z') || ((int)text[i] >= 'A' && (int)text[i] <= 'Z'))
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            words++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentances++;
        }
    }
    
    //get key variables to calculate index
    float L = (100 * (float) letters / words);
    float S = (100 * (float) sentances / words);
    
    float index = 0.0588 * L - 0.296 * S - 15.8;
    
    //outputs. 3 cases. One will run no matter what happens
    if (index < 16 && index >= 0.5)
    {
        printf("Grade %i\n", (int)round(index));
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Before Grade 1\n");
    }


    
    
}

File: pset6/readability/readability.py
from cs50 import get_string


def main():
    
    # first get the text to be graded
    text = get_string("Text: \n")
    
    # the data we need to calculate    
    letters = 0
    words = 1
    sentances = 0
    length = 0
    
    for i in text:
        length += 1

    # count the number of ascii letters in the text
    for i in text:
        if ((ord(i) > 64 and ord(i) < 123) and ord(i) != 96):
            letters += 1
    
    for i in range(length):
        
        # count the number of spaces that are preceded by a letter, i.e words

        if (ord(text[i]) == 32):
            words += 1
        
        # count the number sentances by detecting full stops
        
        if (text[i] == '.' or text[i] == '!' or text[i] == '?'):
            sentances += 1
    
    L = (letters / words) * 100
    S = (sentances / words) * 100
    
    index = 0.0588 * L - 0.296 * S - 15.8
    index = round(index, 0)
    index = int(index)
    if (index < 1):
        print("Before Grade 1")
    elif (index >= 16):
        print("Grade 16+")
    else:
        print(f"Grade: {index}")
    
    
main()
